The other test cases catch this in the decorator. This is caught in the
function body, but since stream() returns a generator, it's not
evaluated until its .next() method is invoked.

=== modified file 'tests/object_test.py'
Index: b/tests/object_test.py
===================================================================
--- a/tests/object_test.py	2009-10-20 16:47:14.000000000 +0200
+++ b/tests/object_test.py	2009-11-02 16:36:10.779219619 +0100
@@ -123,7 +123,7 @@
 
         obj.name = 'a'*(object_name_limit+1) # too-long string
         self.assertRaises(InvalidObjectName, obj.read)
-        self.assertRaises(InvalidObjectName, obj.stream)
+        self.assertRaises(InvalidObjectName, obj.stream().next)
         self.assertRaises(InvalidObjectName, obj.sync_metadata)
         self.assertRaises(InvalidObjectName, obj.write, '')
 
